### ISSUE TITLE
Store key derivation function used and iteration count in payload

### ISSUE THREAD

**Issue by federicobond** [MetaMask/browser-passworder#1]
The KDF used and iteration count should be included in the payload returned from `encrypt` to facilitate upgrades over time. This makes it possible to change the work factor or the algorithm used without breaking backward compatibility.

Library users should be encouraged to upgrade encrypted ciphertexts from time to time to take advantage of this too.

**Comment by danfinlay** [comment]
I agree, and plan to do this eventually, maybe using [pojo-migrator](https://www.npmjs.com/package/pojo-migrator) to allow users of the old format to experience no breaking changes.

In the meanwhile, I wouldn't object to a PR that does this, especially if it had a non-breaking migration process from the old format.

**Comment by federicobond** [comment]
We can always assume the current values if there aren't any defined, to preserve backwards compatibility.

**Comment by danfinlay** [comment]
Exactly.
